parser/parser.y

```diff
@@ -1799,14 +1810,29 @@ DirectResourceGroupRunawayOption:
 	{
 		$$ = &ast.ResourceGroupRunawayOption{Tp: ast.RunawayAction, IntValue: $3.(int32)}
 	}
-|	"WATCH" EqOpt ResourceGroupRunawayWatchOption "DURATION" EqOpt stringLit
+|	"WATCH" EqOpt ResourceGroupRunawayWatchOption WatchDurationOption
 	{
-		_, err := time.ParseDuration($6)
-		if err != nil {
-			yylex.AppendError(yylex.Errorf("The WATCH DURATION option is not a valid duration: %s", err.Error()))
-			return 1
+		if len($4.(string)) > 0 {
+			_, err := time.ParseDuration($4.(string))
+			if err != nil {
+				yylex.AppendError(yylex.Errorf("The WATCH DURATION option is not a valid duration: %s", err.Error()))
+				return 1
+			}
 		}
-		$$ = &ast.ResourceGroupRunawayOption{Tp: ast.RunawayWatch, StrValue: $6, IntValue: $3.(int32)}
+		$$ = &ast.ResourceGroupRunawayOption{Tp: ast.RunawayWatch, StrValue: $4.(string), IntValue: $3.(int32)}
+	}
+
+WatchDurationOption:
+	{
+		$$ = ""
+	}
+|	"DURATION" EqOpt stringLit
+	{
+		$$ = $3
+	}
+|	"DURATION" EqOpt "UNLIMITED"
```

Comment:
I would like to also support string literal "unlimited" as unlimited because it's more intuitive to wrap it into a string instead of a keyword.

Reply:
done